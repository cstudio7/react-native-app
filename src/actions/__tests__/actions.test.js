import {
  loadNames,
  favorite,
  listScreenView,
  favoritesScreenView,
  scrollEvent,
  openNameScreen,
  showRateAppAlert,
  rateApp,
  updateUserVisits,
  resetUserVisits
} from '../actions';


it('creates a LISTSCREEN_NAME_UNFAVORITE action', () => {
  const payload = {
    gender: 'male',
    name: {
      name: 'artur',
      isFavorite: false
    }
  };
  const screen = undefined;
  expect(favorite(screen, payload)).toMatchSnapshot();
});

it('creates a LISTSCREEN_VIEW action', () => {
  expect(favoritesScreenView()).toMatchSnapshot();
});

it('creates a FAVSCREEN_VIEW action', () => {
  expect(listScreenView()).toMatchSnapshot();
});

it('creates a LISTSCREEN_SCROLL action', () => {
  expect(scrollEvent()).toMatchSnapshot();
});

it('creates a FAVSCREEN_SCROLL action', () => {
  expect(scrollEvent('Favorites')).toMatchSnapshot();
});

it('creates a LISTSCREEN_NAME_OPEN action', () => {
  expect(openNameScreen()).toMatchSnapshot();
});

it('creates a FAVSCREEN_NAME_OPEN action', () => {
  expect(openNameScreen('Favorites')).toMatchSnapshot();
});

it('creates an action to load names', () => {
  expect(loadNames()).toMatchSnapshot();
});

it('creates a LISTSCREEN_RATEALERT_SHOW action', () => {
  expect(showRateAppAlert()).toMatchSnapshot();
});

it('creates a LISTSCREEN_RATEALERT_RATE action', () => {
  expect(rateApp()).toMatchSnapshot();
});

it('creates a LISTSCREEN_RATEALERT_UPDATEVISITS action', () => {
  expect(updateUserVisits()).toMatchSnapshot();
});

it('creates a LISTSCREEN_RATEALERT_RESETVISITS action', () => {
  expect(resetUserVisits()).toMatchSnapshot();
});
