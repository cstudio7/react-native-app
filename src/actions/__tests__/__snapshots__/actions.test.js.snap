// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a FAVSCREEN_NAME_OPEN action 1`] = `
Object {
  "type": "FAVSCREEN_NAME_OPEN",
}
`;

exports[`creates a FAVSCREEN_SCROLL action 1`] = `
Object {
  "type": "FAVSCREEN_SCROLL",
}
`;

exports[`creates a FAVSCREEN_VIEW action 1`] = `
Object {
  "type": "LISTSCREEN_VIEW",
}
`;

exports[`creates a LISTSCREEN_NAME_OPEN action 1`] = `
Object {
  "type": "LISTSCREEN_NAME_OPEN",
}
`;

exports[`creates a LISTSCREEN_NAME_UNFAVORITE action 1`] = `
Object {
  "payload": Object {
    "gender": "male",
    "name": Object {
      "isFavorite": false,
      "name": "artur",
    },
  },
  "type": "LISTSCREEN_NAME_UNFAVORITE",
}
`;

exports[`creates a LISTSCREEN_RATEALERT_RATE action 1`] = `
Object {
  "type": "LISTSCREEN_RATEALERT_RATE",
}
`;

exports[`creates a LISTSCREEN_RATEALERT_RESETVISITS action 1`] = `
Object {
  "type": "LISTSCREEN_RATEALERT_RESETVISITS",
}
`;

exports[`creates a LISTSCREEN_RATEALERT_SHOW action 1`] = `
Object {
  "type": "LISTSCREEN_RATEALERT_SHOW",
}
`;

exports[`creates a LISTSCREEN_RATEALERT_UPDATEVISITS action 1`] = `
Object {
  "type": "LISTSCREEN_RATEALERT_UPDATEVISITS",
}
`;

exports[`creates a LISTSCREEN_SCROLL action 1`] = `
Object {
  "type": "LISTSCREEN_SCROLL",
}
`;

exports[`creates a LISTSCREEN_VIEW action 1`] = `
Object {
  "type": "FAVSCREEN_VIEW",
}
`;

exports[`creates an action to load names 1`] = `
Object {
  "type": "FETCH_NAMES_REQUEST",
}
`;
